src/components/toolbar.tsx

```diff
@@ -19,9 +20,14 @@ import "./toolbar.scss";
 // (See `annotation-toolbar.tsx`)
 const ignoredButtons = [ "sparrow", "hide-annotations" ];
 
+// This toolbar works with both the document and section models.
+// Since many of the tools are shared between the two, each model is
+// passed as an optional prop instead of using two separate components.
 interface IProps extends IBaseProps {
-  document: DocumentModelType;
+  document?: DocumentModelType;
+  section?: SectionModelType;
   toolbarModel: IToolbarModel;
+  disabledToolIds?: string[];
```

Comment:
Could the enabled/disabled status be locally controlled by the button itself, rather than a level up on the toolbar?

Reply:
I tried that but had an issue with trying to attach a callback to a mobx-tree created object.  The docs says `type.freeze()` should let you pass a callback but I was getting a runtime error.  This will be more flexible anyway as the select all tool for example would be disabled in two different ways depending if we are viewing a section or a document.